lib/skate/open_route_service_api/directions_request.ex

```diff
@@ -4,13 +4,46 @@ defmodule Skate.OpenRouteServiceAPI.DirectionsRequest do
   """
   @derive Jason.Encoder
 
+  defmodule Options do
+    @moduledoc false
+    defmodule ProfileParams do
+      @moduledoc false
+      defmodule HgvRestrictions do
+        @moduledoc false
+        @type t :: %{length: float(), width: float(), height: float()}
+        defstruct [:length, :width, :height]
+
+        def bus_40ft,
+          do: %{
+            length: 12.192,
+            width: 3.2004,
+            height: 3.5052
+          }
+      end
+
+      @type t :: %{restrictions: HgvRestrictions.t()}
+      defstruct [:restrictions]
+    end
+
+    @type t :: %{profile_params: ProfileParams.t()}
+    defstruct [:profile_params]
+  end
+
```

Comment:
I thought doing this inline would be easy, but it doesn't look great. I'm open to other options for getting this typed and having a good place for bus profiles to live. I personally liked having all this in a single file, but I could be convinced to move these into separate files.
One issue, I had to move the definitions to before their first use, otherwise the compiler complains.